### DISCUSSION TITLE
Query Status changes BEFORE onSuccess/onError/onSettled promises are resolved as of react-query@v2

### DISCUSSION THREAD

**Discussion by jrmyio** [TanStack/query#1029] [General]
It seems that since this commit:
https://github.com/tannerlinsley/react-query/commit/26844a9bc72d162790f7533a4b9d833cdbcf2f2e

Either by accident, or on purpose, the resolve state is triggered BEFORE onSuccess/onError and onSettled are resolved.

This means that when using queryCache.refetchQueries inside a list will still have a deleted item after the loading state is done.

Was this change intentional or should I file an issue?

Should these lines be flipped?
https://github.com/tannerlinsley/react-query/blob/master/src/react/useMutation.ts#L151-L158

as they are in 

https://github.com/tannerlinsley/react-query/blob/master/src/react/useMutation.ts#L151-L158

**Comment by jrmyio** [comment]
Should I make an issue for this, anyone ?

**Reply by boschni** [reply]
Hi @ConneXNL! This has been corrected in V3 `react-query@next`
